src/Shell.cpp

```diff
@@ -0,0 +1,101 @@
+#include "Shell.h"
+#include <iostream>
+#include <sstream>
+#include <fstream>
+#include <unistd.h>
+#include <sys/wait.h>
+
+void Shell::run() {
+    while (true) {
+        std::string command;
+        std::cout << "shell> ";
+        std::getline(std::cin, command);
+        if (command == "exit") {
+            break;
+        }
+        else if (command == "myhistory") {
+            showHistory();
+        }
+        else if (command == "myjobs") {
+            processManager.showJobs();
+        }
+        else {
+            bool isBackground = isBackgroundCommand(command);
+            addToHistory(command);
+            execute(command, isBackground);
+        }
+    }
+}
+
+void Shell::addToHistory(const std::string& command) {
+    commandHistory.push_back(command);
```

Comment:
You were asked to save the history to a file, that way the history is persistent and isn't lost when the program ends. Also, you could fire up a new shell and all your history would be there because it was saved on disk